code/modules/hydroponics/grown/misc.dm

```diff
@@ -244,3 +244,283 @@
+			user.show_message("<span class='notice'>You add a fuse to the coconut!</span>", 1)
+			W.use(1)
+			fused = TRUE
+			icon_state = "coconut_grenade"
+			desc = "A makeshift bomb made out of a coconut. You estimate the fuse is long enough for 5 seconds."
+			name = "coconut bomb"
+			return
+	//ADDING STRAW LOGIC
+	if (istype(W,/obj/item/stack/sheet/mineral/bamboo) && opened == TRUE && straw == FALSE && fused == FALSE)
+		user.show_message("<span class='notice'>You add a bamboo straw to the coconut!</span>", 1)
+		straw = TRUE
+		W.use(1)
+		icon_state += "_straw"
+		desc = "You can already feel like you're on a tropical vacation."
+	//OPENING THE NUT LOGIC
+	if (carved == FALSE && chopped == FALSE)
+		if(W.tool_behaviour == TOOL_SCREWDRIVER)
+			user.show_message("<span class='notice'>You make a hole in the coconut!</span>", 1)
+			carved = TRUE
+			opened = TRUE
+			reagent_flags = OPENCONTAINER
+			ENABLE_BITFIELD(reagents.reagents_holder_flags, OPENCONTAINER)
+			icon_state = "coconut_carved"
+			desc = "A coconut. This one's got a hole in it."
+			playsound(user, W.hitsound, 50, 1, -1)
+			return
+		else if(W.sharpness)
+			user.show_message("<span class='notice'>You slice the coconut open!</span>", 1)
+			chopped = TRUE
+			opened = TRUE
+			reagent_flags = OPENCONTAINER
+			ENABLE_BITFIELD(reagents.reagents_holder_flags, OPENCONTAINER)
+			spillable = TRUE
+			icon_state = "coconut_chopped"
+			desc = "A coconut. This one's sliced open, with all its delicious contents for your eyes to savour."
+			playsound(user, W.hitsound, 50, 1, -1)
+			return
+
+/obj/item/reagent_containers/food/snacks/grown/coconut/attack(mob/living/M, mob/user, obj/target)
+	if(M && user.a_intent == INTENT_HARM && spillable == FALSE)
+		var/obj/item/bodypart/affecting = user.zone_selected //Find what the player is aiming at
+		if (affecting == BODY_ZONE_HEAD && prob(15))
+			//smash the nut open
+			var/armor_block = min(90, M.run_armor_check(affecting, "melee", null, null,armour_penetration)) // For normal attack damage
+			M.apply_damage(force, BRUTE, affecting, armor_block)
+
+			//Sound
+			playsound(user, hitsound, 100, 1, -1)
+
+			//Attack logs
+			log_combat(user, M, "attacked", src)
+
+			//Display an attack message.
+			if(M != user)
+				M.visible_message("<span class='danger'>[user] has cracked open a [src.name] on [M]'s head!</span>", \
+						"<span class='userdanger'>[user] has cracked open a [src.name] on [M]'s head!</span>")
+			else
+				user.visible_message("<span class='danger'>[M] cracks open a [src.name] on their [M.p_them()] head!</span>", \
+						"<span class='userdanger'>[M] cracks open a [src.name] on [M.p_their()] head!</span>")
+
+			//The coconut breaks open so splash its reagents
+			spillable = TRUE
+			SplashReagents(M)
+
+			//Lastly we remove the nut
+			qdel(src)
+		else
+			. = ..()
+		return
+
+	if(fusedactive)
+		return
+
+	if(opened == FALSE)
+		return
+
+	if(!canconsume(M, user))
+		return
```

Comment:
This should be moved to the feeding attempt and shouldn't prevent spessmen from splashing acid filled coconuts onto each other.